pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from collections import OrderedDict
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        "Summary",
+        "Parameters",
+        "Attributes",
+        "Methods",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "Raises",
+        "Warns",
+        "Warnings",
+        "See Also",
+        "Notes",
+        "References",
+        "Examples",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Sort the numpydoc sections into the numpydoc order."""
+        for sec_name in reversed(self.numpydoc_section_order):
+            try:
+                sections.move_to_end(sec_name, last=False)
+            except KeyError:
+                pass
+        return sections
+
+
+class NumpydocNameColonTypeFormatter(NumpydocSectionFormatter):
+    """Ensure proper spacing around the colon separating names from types."""
+
+    name = "numpydoc-name-type-spacing"
+
+    numpydoc_sections_with_parameters = (
+        "Parameters",
+        "Attributes",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "See Also",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Ensure proper spacing around the colon separating names from types."""
+        for section_name, section_lines in sections.items():
+            if section_name in self.numpydoc_sections_with_parameters:
+                # Any section that gets here has a line of hyphens
+                initial_indent = section_lines[1].index("-")
+                for index, line in enumerate(section_lines):
+                    if (
+                        ":" in line
+                        and len(line) > initial_indent
+                        and not line[initial_indent].isspace()
```

Comment:
What is this line checking for?

Reply:
A parameter description might look like

```python
    """Summarize.

    Parameters
    ----------
    x : float
        The x-coordinate: the argument of the function

        New Paragraph: Usually in radians.

    y: float
        The y-coordinate: the abscissa of the plot

    Returns
    """
```
If the line is empty, or just the standard indent for the docstring, there's nothing to format.  
If the line is indented more than the standard, it's a description, and not relevant to this formatter.
If there is no colon, the name or type is missing, and there is also nothing for the formatter to do.

In other words, the lines relevant to this formatter must 
1. have content,
2. be indented flush with the heading, 
3. have content after that indent, and
4. have both a name and a type, separated by a colon

Reply:
I think a concise comment explaining this would be helpful for future us 😄